src/DevYeah.LMS/DevYeah.LMS.Data/CourseRepository.cs

```diff
@@ -9,5 +12,33 @@ public class CourseRepository : Repository<Course>, ICourseRepository
         public CourseRepository(DbContext dbContext) : base(dbContext)
         {
         }
+
+        public IEnumerable<Course> GetAllCourses() => FindAll(course => true);
+
+        public IEnumerable<Course> GetCoursesByCategory(Guid catId)
+        {
+            var dbCtx = GetDbContext();
+            var coursesFilterByCat = dbCtx.Set<Category>()
+                .Include(t => t.CourseCategory)
+                    .ThenInclude(c => c.Course)
+                        .ThenInclude(c => c.Instructor)
+                .Where(t => t.Id == catId)
+                .AsNoTracking()
+                .FirstOrDefault()
```

Comment:
Using `.FirstOrDefault()`  could return null if it is reference type. So .CourseCategory can be null, you might want to use null propagation: ?.